src/annotator/components/ContentInfoBanner.js

```diff
@@ -8,39 +15,108 @@ import { LabeledButton, Link } from '@hypothesis/frontend-shared';
  * A banner that displays information about the current document and the entity
  * that is providing access to it (eg. JSTOR).
  *
+ * layout columns:
+ *  - Logo
+ *  - Container title (only shown on screens at `2xl` breakpoint and wider)
+ *  - Item title with previous and next links
+ *
  * @param {object} props
  *   @param {ContentInfoConfig} props.info
- *   @param {() => void} props.onClose
  */
-export default function ContentInfoBanner({ info, onClose }) {
+export default function ContentInfoBanner({ info }) {
   return (
-    <div className="flex items-center border-b gap-x-4 px-2 py-1 bg-white text-annotator-lg">
-      <Link href={info.logo.link} target="_blank" data-testid="logo-link">
-        <img src={info.logo.logo} alt={info.logo.title} />
-      </Link>
-      <div className="grow">
-        {info.links.previousItem && (
-          <Link href={info.links.previousItem} target="_blank">
-            Previous article
+    <div
+      className={classnames(
+        'h-10 bg-white px-4 text-slate-7 text-annotator-base border-b',
+        'grid items-center',
+        // Two columns in narrower viewports; three in wider
+        'grid-cols-[100px_minmax(0,auto)] gap-x-4',
+        '2xl:grid-cols-[100px_minmax(0,auto)_minmax(0,auto)] 2xl:gap-x-3'
+      )}
+    >
+      <div data-testid="content-logo">
+        {info.logo && (
+          <Link href="https://jstor.org" target="_blank">
+            <img alt={info.logo.title} src={info.logo.logo} />
           </Link>
         )}
-        {info.item.title}
-        {info.item.containerTitle && (
-          <span>
-            {' '}
-            from <i>{info.item.containerTitle}</i>
-          </span>
-        )}
-        {info.links.nextItem && (
-          <Link href={info.links.nextItem} target="_blank">
-            Next article
-          </Link>
+      </div>
+      <div
+        className={classnames(
+          // Container title (this element) is not shown on narrow screens
+          'hidden',
+          '2xl:block 2xl:whitespace-nowrap 2xl:overflow-hidden 2xl:text-ellipsis'
         )}
+        data-testid="content-container-info"
+      >
+        <span className="font-semibold" title={info.item.containerTitle}>
+          {info.item.containerTitle}
```

Comment:
I don't think we explicitly discussed whether the container label should be a link, or whether we even have a link that we could use for that. So there is nothing to do for now. It might come up in future though.